x/params/store/store.go

```diff
@@ -0,0 +1,177 @@
+package store
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Individual parameter store for each keeper
+type Store struct {
+	cdc  *codec.Codec
+	key  sdk.StoreKey
+	tkey sdk.StoreKey
+
+	space []byte
+}
+
+// NewStore constructs a store with namestore
+func NewStore(cdc *codec.Codec, key sdk.StoreKey, tkey sdk.StoreKey, space string) Store {
+	return Store{
+		cdc:  cdc,
+		key:  key,
+		tkey: tkey,
+
+		space: []byte(space),
+	}
+}
+
+// Returns a KVStore identical with ctx,TransientStore(s.key).Prefix()
+func (s Store) kvStore(ctx sdk.Context) sdk.KVStore {
+	return ctx.KVStore(s.key).Prefix(append(s.space, '/'))
+}
+
+// Returns a KVStore identical with ctx.TransientStore(s.tkey).Prefix()
+func (s Store) transientStore(ctx sdk.Context) sdk.KVStore {
+	return ctx.TransientStore(s.tkey).Prefix(append(s.space, '/'))
+}
+
+// Get parameter from store
+func (s Store) Get(ctx sdk.Context, key string, ptr interface{}) {
```

Comment:
Why do all these functions take `string` then convert to `[]byte` instead of taking `[]byte` directly? We shouldn't be passing keys around as strings in most cases.